proto/osmosis/tokenfactory/v1beta1/genesis.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+package osmosis.tokenfactory.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "osmosis/tokenfactory/v1beta1/authorityMetadata.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v7/x/tokenfactory/types";
+
+// GenesisState defines the tokenfactory module's genesis state.
+message GenesisState {
+  option (gogoproto.equal) = true;
+
+  repeated GenesisDenom factory_denoms = 1 [
+    (gogoproto.moretags) = "yaml:\"factory_denoms\"",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message GenesisDenom {
+  option (gogoproto.equal) = true;
+
+  string denom = 1 [ (gogoproto.moretags) = "yaml:\"denom\"" ];
+  DenomAuthorityMetadata authority_metadata = 2 [
+    (gogoproto.moretags) = "yaml:\"authority_metadata\"",
+    (gogoproto.nullable) = false
+  ];
+}
```

Comment:
Why is this type called genesis denom? Presumably this is whats going to get stored in state for every created denom (as the admin metadata has to be maintained somewhere)

Reply:
Perhaps it should be called FactoryDenom?

Reply:
It's only the type that's stored in the genesis file.  In state, `denom` is a key and just `DenomAuthorityMetadata` is serialized and stored in state